libsolidity/ast/Types.cpp

```diff
@@ -1876,6 +1886,15 @@ std::unique_ptr<ReferenceType> ArrayType::copyForLocation(DataLocation _location
 	return copy;
 }
 
+std::unique_ptr<ReferenceType> ArraySliceType::copyForLocation(DataLocation _location, bool _isPointer) const
+{
+	std::unique_ptr<ArrayType> arrayCopy(static_cast<ArrayType*>(m_arrayType.copyForLocation(_location, _isPointer).release()));
+	auto copy = make_unique<ArraySliceType>(*arrayCopy.get());
```

Comment:
ArrayType is saved as reference in ArraySliceType, I believe this will produce bugs as arrayCopy is local object which will be destroyed after returning from function...

Reply:
Not sure how to implement this function correctly...

Reply:
Hm, maybe ``ArrayType::copyForLocation`` should actually create the array type using the ``TypeProvider`` and return a plain pointer - then we have guaranteed safe life times...
Why do you need ``ArraySliceType::copyForLocation`` at all, though? What calls this?

Reply:
It was asserting there when I started doing .length, now I removed it as it is not needed